meetings/working-groups/collection-literals/CL-LDM-2023-08-14.md

```diff
@@ -0,0 +1,205 @@
+
+*See proposal: [Overload resolution](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#overload-resolution)*
+
+For overload resolution, we've split the possible target types into three groups, in order of preference:
+1. span types
+1. arrays and concrete types
+1. interfaces
+
+Spans are preferred over non-spans.
+```c#
+// Example 1: Span<T> and T[]
+SpanAndArray([1, 2, 3]);         // Span<int>
+SpanAndArray(new[] { 1, 2, 3 }); // int[]
+
+static void SpanAndArray<T>(Span<T> args) { }
+static void SpanAndArray<T>(T[] args) { }
+```
+
+Arrays and concrete types are preferred over interfaces.
+```c#
+// Example 2: params object[] and IEnumerable<object>
+ParamsAndEnumerable([1, 2, 3]); // params object[]
+
+static void ParamsAndEnumerable(params object[] args) { }
+static void ParamsAndEnumerable(IEnumerable<object> args) { }
+```
+
+There is no preference between types that are neither spans nor interfaces.
+```c#
+// Example 3: MyCollection<T> and T[]
+MyCollectionAndArray([1, 2, 3]);  // ambiguous
+
+static void MyCollectionAndArray<T>(MyCollection<T> args) { }
+static void MyCollectionAndArray<T>(T[] args) { }
+```
+
+A single rule is added to [*better conversion from expression*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#11644-better-conversion-from-expression) to handle the preference order above: see [proposal](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#overload-resolution).
+
+The added rule applies to *collection expression arguments only*.
+* The rule is not a breaking change to existing code.
+* Changing grouping is a *breaking change*.
+* Adding target types is a *breaking change*.
+
+**Question**: Should the first group include all `ref struct` types, not just spans?
+
+**Question**: Should we differentiate between arrays and concrete types? If so, how?
+
+*An array or concrete type is already preferred over an interface if the array or concrete type implements that interface. And if the array or concrete type does not implement the interface, we shouldn't prefer one over the other. Instead, the overload resolution proposal should be: Prefer a span over an array or interface if the iteration type of the span is implicitly convertible to the iteration type of the array or interface.*
+
+## Type inference
+
+*See proposal: [Type inference](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#type-inference)*
+
+Method type inference involving collection expressions allows inferring the [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) of a collection type. There is no support for inferring the containing *collection type*.
+
+The *iteration type* is defined by `foreach`. If a *collection initializer type* implements `IEnumerable` only, the iteration type is `object`.
```

Comment:
Being super nitpicky, a counterexample is where the type implements `IEnumerable` explicitly and also has a GetEnumerator method (which doesn't implement any interface method) which returns a different enumerator type with a different `foreach` element type.